src/tests/test_bootstrap_in_place.py

```diff
@@ -237,11 +255,150 @@ def test_bootstrap_in_place(
         self.download_live_image(
             f"{BUILD_DIR}/installer-image.iso", extract_rhcos_url_from_ocp_installer(INSTALLER_BINARY)
         )
-        image_path = self.embed("installer-image.iso", "bootstrap-in-place-for-live-iso.ign", EMBED_IMAGE_NAME)
+        return self.embed("installer-image.iso", "bootstrap-in-place-for-live-iso.ign", EMBED_IMAGE_NAME)
+
+    @JunitTestSuite()
+    def test_bootstrap_in_place(
+        self, controller: NodeController, controller_configuration: BaseNodeConfig, cluster_configuration: ClusterConfig
+    ):
+        image_path = self.prepare_installation(controller_configuration, cluster_configuration)
 
         log.info("Starting node...")
         cluster_configuration.iso_download_path = image_path
         controller.start_all_nodes()
         log.info("Node started!")
 
         self.waiting_for_installation_completion(controller, cluster_configuration)
+
+
+class TestBootstrapInPlaceWorker(TestBootstrapInPlace):
+    @pytest.fixture
+    def new_controller_configuration(self, request) -> BaseNodeConfig:
+        return TerraformConfig(
+            masters_count=1,
+            workers_count=1,
+            master_memory=16 * 1024,  # in megabytes
+            master_vcpu=16,
+            worker_memory=16 * 1024,  # in megabytes
+            worker_vcpu=16,
+            bootstrap_in_place=True,
+            running=False,
+        )
+
+    def render_worker_live_iso_ignition(self, install_device: str):
+        """
+        The worker live iso ignition file is embedded in the live ISO for the worker
+        and is responsible for:
+            - Copying the worker.ign file into the live filesystem
+            - Creating a one-shot systemd unit service which runs coreos-installer with the worker.ign
+            - Rebooting the node once the operating system has been written to disk
+
+        The worker then starts with the installed RHCOS+worker.ign and attempts to join the cluster
+
+        @param install_device The path of the disk to install RHCOS on (e.g. /dev/vda)
+        """
+        with open(os.path.join(RESOURCES_DIR, WORKER_LIVE_IGNITION_TEMPLATE), "r") as f:
+            live_iso_ignition_template_contents = f.read()
+
+        with open(os.path.join(RESOURCES_DIR, WORKER_INSTALL_SCRIPT), "rb") as f:
+            worker_install_script_contents = f.read()
+
+        with open(os.path.join(IBIP_DIR, "worker.ign"), "rb") as f:
```

Comment:
In case this fails due to FileNotFoundError, log that this file is generated in 4.11 release and above